src/WaCore.Crud.Services/WacListDataService.cs

```diff
@@ -1,23 +1,45 @@
 ﻿using System.Collections.Generic;
+using System.Threading.Tasks;
 using WaCore.Crud.Contracts.Data;
+using WaCore.Crud.Contracts.Dtos;
 using WaCore.Crud.Contracts.Services;
+using WaCore.Crud.Contracts.Utils;
+using WaCore.Crud.Utils;
 
 namespace WaCore.Crud.Services
 {
     public abstract class WacListDataService<TEntity, TFilter, TDto> : IWacListDataService<TFilter, TDto>
+        where TFilter : IWacFilter
     {
-        private readonly IWacListDataRepository<TEntity, TFilter> repo;
+        protected readonly IWacListDataRepository<TEntity, TFilter> repo;
 
         public WacListDataService(IWacListDataRepository<TEntity, TFilter> repo)
         {
             this.repo = repo;
```

Comment:
Can you please rename it to Repo and remove this?

Reply:
yep, done